asserts/system_user.go

```diff
@@ -28,7 +28,9 @@ import (
 	"time"
 )
 
-var validSystemUserUsernames = regexp.MustCompile(`^[a-z0-9][-a-z0-9+.-_]*$`)
+// validSystemUserUsernames matches very closely the regex we allow by osutil/user.go:IsValidUsername
+// however this regex is a tiny bit less strict by also allowing '-' in the username.
```

Comment:
I don't understand the comment about "-",  it's also in the other regexp:

```
^[a-z0-9][-a-z0-9._]*$
```
there's  a `-` in `[-`

But  this regexp has also been quite buggy because of the `-` in `.-_`  that means all chars in between those two, that `-` needs to be removed.

Reply:
hm so technically `f[^^]` is a valid username, fun

Reply:
Fixed this! Seems I misread the regex :-)